libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -425,41 +432,42 @@ void AP_ESC_Telem::update_telem_data(const uint8_t esc_index, const AP_ESC_Telem
     }
 
     _have_data = true;
+    volatile AP_ESC_Telem_Backend::TelemetryData &telemdata = _telem_data[esc_index];
 
 #if AP_TEMPERATURE_SENSOR_ENABLED
     // always allow external data. Block "internal" if external has ever its ever been set externally then ignore normal "internal" updates
     const bool has_temperature = (data_mask & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE_EXTERNAL) ||
-        ((data_mask & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE) && !(_telem_data[esc_index].types & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE_EXTERNAL));
+        ((data_mask & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE) && !(telemdata.types & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE_EXTERNAL));
 
     const bool has_motor_temperature = (data_mask & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE_EXTERNAL) ||
-        ((data_mask & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE) && !(_telem_data[esc_index].types & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE_EXTERNAL));
+        ((data_mask & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE) && !(telemdata.types & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE_EXTERNAL));
 #else
     const bool has_temperature = (data_mask & AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE);
     const bool has_motor_temperature = (data_mask & AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE);
 #endif
 
     if (has_temperature) {
-        _telem_data[esc_index].temperature_cdeg = new_data.temperature_cdeg;
+        telemdata.temperature_cdeg = new_data.temperature_cdeg;
     }
     if (has_motor_temperature) {
-        _telem_data[esc_index].motor_temp_cdeg = new_data.motor_temp_cdeg;
+        telemdata.motor_temp_cdeg = new_data.motor_temp_cdeg;
     }
     if (data_mask & AP_ESC_Telem_Backend::TelemetryType::VOLTAGE) {
-        _telem_data[esc_index].voltage = new_data.voltage;
+        telemdata.voltage = new_data.voltage;
     }
     if (data_mask & AP_ESC_Telem_Backend::TelemetryType::CURRENT) {
-        _telem_data[esc_index].current = new_data.current;
+        telemdata.current = new_data.current;
     }
     if (data_mask & AP_ESC_Telem_Backend::TelemetryType::CONSUMPTION) {
-        _telem_data[esc_index].consumption_mah = new_data.consumption_mah;
+        telemdata.consumption_mah = new_data.consumption_mah;
     }
     if (data_mask & AP_ESC_Telem_Backend::TelemetryType::USAGE) {
-        _telem_data[esc_index].usage_s = new_data.usage_s;
+        telemdata.usage_s = new_data.usage_s;
     }
 
-    _telem_data[esc_index].count++;
-    _telem_data[esc_index].types |= data_mask;
-    _telem_data[esc_index].last_update_ms = AP_HAL::millis();
+    ++telemdata.count;
```

Comment:
This is also contrary to the vast majority of our code.

Reply:
Here I agree, I will fix this in the next force push.